packages/nextjs/src/config/loaders/dataFetchersLoader.ts

```diff
@@ -94,44 +105,73 @@ function wrapFunctions(userCode: string, templateCode: string, filepath: string)
-
   // For now this loader only works for ESM code
   if (!isESM(userCode)) {
     return userCode;
   }
 
-  // If none of the functions we want to wrap appears in the page's code, there's nothing to do. (Note: We do this as a
-  // simple substring match (rather than waiting until we've parsed the code) because it's meant to be an
-  // as-fast-as-possible fail-fast. It's possible for user code to pass this check, even if it contains none of the
-  // functions in question, just by virtue of the correct string having been found, be it in a comment, as part of a
-  // longer variable name, etc. That said, when we actually do the code manipulation we'll be working on the code's AST,
-  // meaning we'll be able to differentiate between code we actually want to change and any false positives which might
-  // come up here.)
-  if (Object.keys(DATA_FETCHING_FUNCTIONS).every(functionName => !userCode.includes(functionName))) {
-    return userCode;
-  }
+  // We know one or the other will be defined, depending on the version of webpack being used
+  // eslint-disable-next-line @typescript-eslint/no-non-null-assertion
+  const { projectDir } = 'getOptions' in this ? this.getOptions() : this.query;
 
-  const templatePath = path.resolve(__dirname, '../templates/dataFetchersLoaderTemplate.js');
-  // make sure the template is included when runing `webpack watch`
-  this.addDependency(templatePath);
+  // Proxy the processed file
+  if (!this.resourceQuery.includes('sentry-proxy-loader')) {
+    const ast = makeAST(userCode, true); // is there a reason to ever parse without typescript?
 
-  const templateCode = fs.readFileSync(templatePath).toString();
+    const exportedIdentifiers = getExportIdentifiers(ast);
 
-  const [modifiedUserCode, modifiedTemplateCode] = wrapFunctions(
-    userCode,
-    templateCode,
-    // Relative path to the page we're currently processing, for use in error messages
-    path.relative(projectDir, this.resourcePath),
-  );
+    let outputFileContent = '';
 
-  // Fill in template placeholders
-  let injectedCode = modifiedTemplateCode;
-  for (const { placeholder, alias } of Object.values(DATA_FETCHING_FUNCTIONS)) {
-    injectedCode = injectedCode.replace(placeholder, alias);
-  }
+    if (exportedIdentifiers.length > 0) {
+      outputFileContent += `export { ${exportedIdentifiers.join(', ')} } from "${
+        this.resourcePath
+      }?sentry-proxy-loader";`;
+    }
 
-  return `${modifiedUserCode}\n${injectedCode}`;
+    if (hasDefaultExport(ast)) {
+      outputFileContent += `
+        import { default as _sentry_default } from "${this.resourcePath}?sentry-proxy-loader";
+        import { withSentryGetInitialProps } from "@sentry/nextjs/build/esm/config/wrappers";
+        Object.defineProperty(
+          _sentry_default,
+          'getInitialProps',
+          { value: withSentryGetInitialProps(_sentry_default.getInitialProps) }
+        );
```

Comment:
Why `defineProperty`? And won't this change it from enumerable to non-enumerable?

Reply:
IIRC there was some situation where when you import something the value is frozen (?). I bet you can remember what it was. I also remember running into a problem testing locally where `default` or rather `_sentry_default` wasn't writable and this fixed it but I can't remember what it was...

Added a line so that we keep the enumerable: 4c97ee9089555051b1500ec8a2928da6889570fd

Reply:
Discussed: Apparently this was something else. Simply assinging should be fine.

Reply:
Implemented in c0577da4ad77711f31e822dd5dce8490a2f59850